awx/api/serializers.py

```diff
@@ -1542,6 +1542,15 @@ def to_representation(self, obj):
     def validate_host_filter(self, host_filter):
         if host_filter:
             try:
+                for match in models.Field.get_lookups().keys():
```

Comment:
This makes it so that if you try to _set_ a `host_filter` in the API like this:

`ansible_facts__ansible_distribution__startswith=Cent`

...you get an HTTP 400.

Reply:
My main question on this PR is, are there no other lookups that a user might attempt than the ones that are registered on `Field`?  It seems vaguely plausible to me that the `IntegerField` inequality operators might be attempted, for example (though maybe I don't have a clear enough mental model yet of what winds up in `ansible_facts`).

Reply:
Huh, never mind, I see that the conversation covered this topic while I was typing.

Reply:
This is a good question.  `awx.main.fields.JSONBField` _does_ have some extra lookups that aren't on the base `Field`:

```python
In [5]: JSONBField.get_lookups().keys() - Field.get_lookups().keys()
Out[5]: {'contained_by', 'has_any_keys', 'has_key', 'has_keys'}
```

Now, `jsonb` *does* have explicit operators for these types of lookups:

https://www.postgresql.org/docs/9.4/functions-json.html#FUNCTIONS-JSONB-OP-TABLE

Maybe we should filter these and add support later, too? I'd say that's also likely an enhancement, though.